html_link_parser.py

```
@@ -0,0 +1,454 @@
+        """Get set of unique href values."""
+        return {link.href for link in self.links}
+    
+    def get_external_domains(self) -> Set[str]:
+        """Get set of external domains referenced."""
+        domains = set()
+        for link in self.external_links:
+            try:
+                parsed = urlparse(link.href)
+                if parsed.netloc:
+                    domains.add(parsed.netloc)
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)